.github/workflows/test_on_push.yml

```diff
@@ -72,61 +114,107 @@ jobs:
         # sometimes gfortran cannot be found, so reinstall gcc just to be sure
         run: |
           brew analytics off
-          brew install graphviz libomp
+          brew install graphviz libomp openblas
           brew reinstall gcc
 
-      - name: Install Windows system dependencies
-        if: matrix.os == 'windows-latest'
-        run: choco install graphviz --version=8.0.5
-
       - name: Set up Python ${{ matrix.python-version }}
         id: setup-python
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
           cache: 'pip'
 
-      - name: Install nox
-        run: python -m pip install nox
+      - name: Install nox and tomlkit
+        run: python -m pip install nox tomlkit
 
-      - name: Cache pybamm-requires nox environment for GNU/Linux and macOS
+      - name: Cache pybamm-requires nox environment for macOS
         uses: actions/cache@v4
-        if: matrix.os != 'windows-latest'
         with:
           path: |
-            # Repository files
-            ${{ github.workspace }}/pybind11/
-            ${{ github.workspace }}/install_KLU_Sundials/
             # Headers and dynamic library files for SuiteSparse and SUNDIALS
-            ${{ env.HOME }}/.local/lib/
-            ${{ env.HOME }}/.local/include/
+            ${{ github.workspace }}/install_KLU_Sundials/
```

Comment:
You can also cache `sundials_KLU_libs/` here and possibly escape the installation step.